cl/people_db/filters.py

```diff
@@ -293,3 +335,16 @@ class Meta:
             "date_modified": DATETIME_LOOKUPS,
             "name": ALL_TEXT_LOOKUPS,
         }
+
+    def filter_roles(self, qs, name, value):
+        if not value:
+            return qs
+
+        role_filters = self.get_filters_for_join_table(name)
+
+        prefetch = Prefetch(
+            name,
+            queryset=Role.objects.filter(**role_filters),
+            to_attr=f"filtered_{name}",
+        )
+        return qs.prefetch_related(prefetch)
```

Comment:
I noticed that this block of code is the same as in `PartyFilter`. Would it be possible to move it to a common place? Maybe within `FilterManyToManyMixin` or as a shared helper method?

Reply:
> Maybe within FilterManyToManyMixin or as a shared helper method?

While refactoring for common logic is generally a good idea, I believe a separate mixin for role filtering would provide a cleaner solution. This would keep the `FilterManyToManyMixin` focused on its core functionality and avoid potential conflicts or overcomplication.